examples/dixonsparseelim.C

```diff
@@ -0,0 +1,214 @@
+/* Copyright (C) The LinBox group
+ * ========LICENCE========
+ * This file is part of the library LinBox.
+ *
+ * LinBox is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+/**\file examples/dixonsparseelim.C
+ @example examples/dixonsparseelim.C
+ @author Jean-Guillaume.Dumas@univ-grenoble-alpes.fr
+ * \brief Dixon System Solving Lifting using sparse LU
+ *        (direct match to dixendenseLU)
+ * \ingroup examples
+ */
+#include <iostream>
+#include <omp.h>
+#include "linbox/matrix/sparse-matrix.h"
+#include "linbox/solutions/solve.h"
+#include "linbox/util/matrix-stream.h"
+#include "linbox/solutions/methods.h"
+#include "linbox/util/error.h"
+
+using namespace LinBox;
+typedef Givaro::ZRing<Givaro::Integer> Ints;
+typedef DenseVector<Ints> ZVector;
+
+
+struct FixPrime {
```

Comment:
same here: factor out this code and adapt it to the PrimeIterator API  in random-prime.h. for instance creating a new `IteratorCategoris::ConstantTag`

Reply:
idem.